cartography/intel/aws/lambda_function.py

```diff
@@ -29,47 +33,193 @@ def get_lambda_data(boto3_session: boto3.session.Session, region: str) -> List[D
+    lambda.signingprofileversionarn = lf.SigningProfileVersionArn,
+    lambda.signingjobarn = lf.SigningJobArn,
     lambda.lastupdated = {aws_update_tag}
-    WITH lambda
+    WITH lambda, lf
     MATCH (owner:AWSAccount{id: {AWS_ACCOUNT_ID}})
     MERGE (owner)-[r:RESOURCE]->(lambda)
     ON CREATE SET r.firstseen = timestamp()
     SET r.lastupdated = {aws_update_tag}
-    WITH lambda
-    MATCH (role:AWSPrincipal{arn: {Role}})
+    WITH lambda, lf
+    MATCH (role:AWSPrincipal{arn: lf.Role})
     MERGE (lambda)-[r:STS_ASSUME_ROLE_ALLOW]->(role)
     ON CREATE SET r.firstseen = timestamp()
     SET r.lastupdated = {aws_update_tag}
     """
 
+    neo4j_session.run(
+        ingest_lambda_functions,
+        lambda_functions_list=data,
+        Region=region,
+        AWS_ACCOUNT_ID=current_aws_account_id,
+        aws_update_tag=aws_update_tag,
+    )
+
+
+@timeit
+def get_function_aliases(lambda_function: Dict, client: botocore.client.BaseClient) -> List[Any]:
+    aliases: List[Any] = []
+    paginator = client.get_paginator('list_aliases')
+    for page in paginator.paginate(FunctionName=lambda_function['FunctionName']):
+        aliases.extend(page['Aliases'])
+
+    return aliases
+
+
+@timeit
+def get_event_source_mappings(lambda_function: Dict, client: botocore.client.BaseClient) -> List[Any]:
+    event_source_mappings: List[Any] = []
+    paginator = client.get_paginator('list_event_source_mappings')
+    for page in paginator.paginate(FunctionName=lambda_function['FunctionName']):
+        event_source_mappings.extend(page['EventSourceMappings'])
+
+    return event_source_mappings
+
+
+@timeit
+def get_lambda_function_details(
+        boto3_session: boto3.session.Session, data: List[Dict], region: str,
+) -> Generator[Any, Any, Any]:
+    client = boto3_session.client('lambda', region_name=region)
     for lambda_function in data:
-        neo4j_session.run(
-            ingest_lambda_functions,
-            LambdaName=lambda_function["FunctionName"],
-            Arn=lambda_function["FunctionArn"],
-            Runtime=lambda_function["Runtime"],
-            Role=lambda_function["Role"],
-            Description=lambda_function["Description"],
-            Timeout=lambda_function["Timeout"],
-            MemorySize=lambda_function["MemorySize"],
-            LastModified=lambda_function["LastModified"],
-            Region=region,
-            AWS_ACCOUNT_ID=current_aws_account_id,
-            aws_update_tag=aws_update_tag,
-        )
+        function_aliases = get_function_aliases(lambda_function, client)
+        event_source_mappings = get_event_source_mappings(lambda_function, client)
+        layers = lambda_function.get('Layers', [])
+        yield lambda_function['FunctionArn'], function_aliases, event_source_mappings, layers
+
+
+@timeit
+def load_lambda_function_details(
+        neo4j_session: neo4j.Session, lambda_function_details: List[Tuple[Any, Any, Any, Any]], update_tag: int,
```

Comment:
Can you be more specific than `Any, Any, Any, Any` in `List[Tuple[Any, Any, Any, Any]]`?

Reply:
Yes, done